core-primitives/stf-executor/src/executor.rs

```diff
@@ -318,17 +320,17 @@ where
 				},
 			};
 
-			// Check time
+			// Break if allowed time window is over.
 			if ends_at < duration_now() {
 				break
 			}
 		}
 
-		self.state_handler
-			.write(state, state_lock, shard)
-			.map_err(|e| Error::StateHandler(e))?;
-
-		Ok(BatchExecutionResult { executed_operations: executed_calls, initial_state_hash })
+		Ok(BatchExecutionResult {
```

Comment:
No state permanent state change, instead, the updated state is now returned within BatchExecutionResult.

Reply:
I thought about that too. But for composing the block, the SidechainDB is used: https://github.com/integritee-network/worker/blob/separate-block-production-from-state-mutation/sidechain/block-composer/src/block_composer.rs#L104

Reply:
But we might think about removing the whole db querying - the block number could be included as a value as well, that would simplify it even more.

Reply:
@clangenb @mullefel your thoughts on this?

Reply:
We can leave it for now IMO, just to not overload this PR even more. I'm not sure it would make our code that much better - I'd have to see it in the greater context, i.e. if it makes more sense to get those pieces of information in the caller.